packages/block-library/src/post-comments-count/edit.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * WordPress dependencies
+ */
+import { useEntityId } from '@wordpress/core-data';
+import { useState, useEffect } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
+import { addQueryArgs } from '@wordpress/url';
+
+function PostCommentsCountDisplay() {
+	const postId = useEntityId( 'postType', 'post' );
+	const [ commentsCount, setCommentsCount ] = useState();
+	useEffect( () => {
+		const currentPostId = postId;
+		apiFetch( {
+			path: addQueryArgs( '/wp/v2/comments', {
+				post: postId,
+			} ),
+			parse: false,
+		} ).then( ( res ) => {
+			// Stale requests will have the `currentPostId` of an older closure.
+			if ( currentPostId === postId ) {
+				setCommentsCount( res.headers.get( 'X-WP-Total' ) );
+			}
+		} );
+	}, [ postId ] );
+	return commentsCount !== undefined && commentsCount;
+}
+
+export default function PostCommentsCountEdit() {
+	if ( ! useEntityId( 'postType', 'post' ) ) {
+		return 'Post Comments Count Placeholder';
+	}
+	return <PostCommentsCountDisplay />;
```

Comment:
do we want to wrap in a span (and use the block className prop) to allow theme styles?

Reply:
This is just a scaffold. The design needs a lot more thought and work.

Reply:
Let's just add a span wrapper for v0 and continue from there.

Reply:
Done

Reply:
Aren't you AFK? :) Go AFK really

Reply:
I am always around :smile: I'm waiting at the airport.